src/test/java/net/rakugakibox/springbootext/logback/access/tomcat/TomcatRequestAttributesTest.java

```diff
@@ -0,0 +1,126 @@
+package net.rakugakibox.springbootext.logback.access.tomcat;
+
+import ch.qos.logback.access.spi.IAccessEvent;
+import net.rakugakibox.springbootext.logback.access.test.AccessEventAssert;
+import net.rakugakibox.springbootext.logback.access.test.SingletonQueueAppender;
+import net.rakugakibox.springbootext.logback.access.test.TextController;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.autoconfigure.EnableAutoConfiguration;
+import org.springframework.boot.autoconfigure.web.EmbeddedServletContainerAutoConfiguration;
+import org.springframework.boot.test.SpringApplicationConfiguration;
+import org.springframework.boot.test.TestRestTemplate;
+import org.springframework.boot.test.WebIntegrationTest;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.Import;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.RequestEntity;
+import org.springframework.http.ResponseEntity;
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+import org.springframework.web.client.RestTemplate;
+import org.springframework.web.util.UriComponentsBuilder;
+
+import java.io.IOException;
+import java.net.URI;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+/**
+ * The test of requestAttributes on Tomcat.
+ */
+@RunWith(SpringJUnit4ClassRunner.class)
+@SpringApplicationConfiguration
+@WebIntegrationTest(
+        value = {"logback.access.config=classpath:logback-access-test.singleton-queue.xml",
+                "server.use-forward-headers=true",
+                // The Proto and Port Header must be explicitly set to work.
+                "server.tomcat.protocol-header=X-Forwarded-Proto",
+                "server.tomcat.port-header=X-Forwarded-Port",
+                "server.tomcat.protocol-header-https-value=https"
```

Comment:
Can I ask you why set explicitly?
I think it is better to use default value.
I passed the test with it.

* https://github.com/spring-projects/spring-boot/blob/v1.3.5.RELEASE/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/web/ServerProperties.java#L783-L784
* https://github.com/spring-projects/spring-boot/blob/v1.3.5.RELEASE/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/web/ServerProperties.java#L583
* https://github.com/spring-projects/spring-boot/blob/v1.3.5.RELEASE/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/web/ServerProperties.java#L578